src/github.com/getlantern/flashlight/client/reverseproxy.go

```diff
@@ -40,23 +28,25 @@ func (client *Client) initReverseProxy(bal *balancer.Balancer, dumpHeaders bool)
 		DisableKeepAlives: true,
 	}
 
+	// Just choose a random dialer that also takes care of things like the
+	// authentication token.
+	dialer := client.getBalancer().RandomDialer()
```

Comment:
Thanks @xiam! I just tweaked things to reuse `balancer.Dial` a bit more. There's a chicken and egg issue where we need to know which connection we're using before knowing which auth token to put in the request while the reverseproxy requires them in the opposite order. This gets around that by storing the connection and then just returning that existing connection in the dial method.

What do you think?